cms/models/pagemodel.py

```diff
@@ -990,7 +990,7 @@ def publish(self, language):
             self.mark_descendants_as_published(language)
 
         if language in self.title_cache:
-            del self.title_cache[language]
+            self._clear_internal_cache()
```

Comment:
This is too heavy, you're effectively dropping the cached node tree and any prefetched objects. I would advise dropping only what is required, which appears to me to be clearing the title cache for the current and fallback language ("en" and "de" in the test). 

Just clear the title cache rather than clearing all caches!
"""
        self.title_cache = {}
"""